exercises/palindrome-products/src/test/java/PalindromeCalculatorTest.java

```diff
@@ -1,25 +1,48 @@
 import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Test;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
 
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.SortedMap;
 import java.util.stream.Collectors;
+import java.util.NoSuchElementException;
+
+import java.lang.IllegalArgumentException;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 
 public class PalindromeCalculatorTest {
     private PalindromeCalculator palindromeCalculator;
+    
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
 
     @Before
     public void setup() {
         palindromeCalculator = new PalindromeCalculator();
     }
+    
+    @Test
+    public void smallestPalindromeFromSingleDigitFactors() {
+        final List<List<Integer>> expected = Collections.unmodifiableList(
```

Comment:
At present, we try to avoid use of the keyword `final` as per our [POLICIES](https://github.com/exercism/java/blob/master/POLICIES.md#avoid-using-final)